corehq/apps/users/models.py

```diff
@@ -1334,6 +1334,13 @@ def set_location(self, location):
         Set the location, and all important user data, for
         the user.
         """
+        if not location:
+            # Remove associated info
+            self.user_data.pop('commtrack-supply-point', None)
+            self.user_data.pop('commcare_primary_case_sharing_id', None)
+            self.location_id = None
+            return
+
```

Comment:
@twymer does this look right to you?  There are some `remove_location` type stuff in this same mixin, but they seemed like they were for a different thing.